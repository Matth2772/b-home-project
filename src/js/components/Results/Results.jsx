import React, { Component } from 'react';
import ListGroup from 'react-bootstrap/Form';
import PropTypes from 'prop-types';
import Result from '../Result/Result';

class Results extends Component {
  constructor() {
    super();
    this.renderResults = this.renderResults.bind(this);
  }

  renderResults() {
    const { results } = this.props;
    return results.map(result => <Result result={result} key={result.objectID} />);
  }

  render() {
    return (
      <ListGroup>
        {this.renderResults()}
      </ListGroup>
    );
  }
}

Results.propTypes = {
  results: PropTypes.arrayOf(PropTypes.object),
};

Results.defaultProps = {
  results: [],
};


export default Results;
