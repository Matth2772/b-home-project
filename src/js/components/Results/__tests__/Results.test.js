/* eslint-disable */
import React from 'react';
import Results from '../Results';
import renderer from 'react-test-renderer';

test('Results rendering well', () => {
  const component = renderer.create(
    <Results/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
