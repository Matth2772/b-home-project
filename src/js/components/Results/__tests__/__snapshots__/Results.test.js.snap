// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results rendering well 1`] = `
<form
  className=""
/>
`;
